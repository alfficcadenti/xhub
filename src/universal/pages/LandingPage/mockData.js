// export const CSRData = [
//     {brandName: 'BEX', CSRTrend: [1, 2, 3, 6, 7, 3, 4, 10], CSRByDevice: {mobile: 90, tablet: 99, desktop: 60}},
//     {brandName: 'Hotels.com', CSRTrend: [1, 2, 3, 6, 5, 11, 10, 6], CSRByDevice: {mobile: 10, tablet: 99, desktop: 60}},
//     {brandName: 'Vrbo', CSRTrend: [10, 9, 3, 4, 5, 6, 4, 9], CSRByDevice: {mobile: 97, tablet: 50, desktop: 60}},
// ];

export const CSRData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 0.99, Vrbo: 0.1, 'Hotels.com': 0.1,
    },
    {
        'time': '2020-05-21T11:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.2,
    },
    {
        'time': '2020-05-21T12:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.3,
    },
    {
        'time': '2020-05-21T13:30:00.000Z', BEX: 0.9, Vrbo: 1, 'Hotels.com': 0.7,
    },
    {
        'time': '2020-05-21T14:30:00.000Z', BEX: 0.9, Vrbo: 0.9, 'Hotels.com': 0.98,
    },
    {
        'time': '2020-05-21T15:30:00.000Z', BEX: 0.5, Vrbo: 0.7, 'Hotels.com': 0.8,
    },
    {
        'time': '2020-05-21T16:30:00.000Z', BEX: 0.99, Vrbo: 0.89, 'Hotels.com': 0.72,
    },
];

export const bookingsData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 4000, Vrbo: 2400, 'Hotels.com': 2400,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3000, Vrbo: 1398, 'Hotels.com': 2210,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2000, Vrbo: 9800, 'Hotels.com': 2290,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2780, Vrbo: 3908, 'Hotels.com': 2000,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 1890, Vrbo: 4800, 'Hotels.com': 2181,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2390, Vrbo: 3800, 'Hotels.com': 2500,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3490, Vrbo: 4300, 'Hotels.com': 2100,
    },
];