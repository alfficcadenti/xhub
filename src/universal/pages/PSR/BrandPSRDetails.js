import React, {PureComponent} from 'react';
import h from './psrHelpers'
import PropTypes from 'prop-types';
import DataTable from '../../components/DataTable/index';
import './styles.less';

class BrandPSRDetails extends PureComponent {
    render() {
        const {data} = this.props;
        const dataForTable = h.formatDataForTable(data);
        return (
            <div id='PSRDetails'>
                <DataTable
                    data={dataForTable}
                    columns={['Line Of Business','Last 24 hours','Last 7 days','Last 28 days']}
                    paginated={false}
                />
            </div>
        );
    }
}

BrandPSRDetails.propTypes = {
    data: PropTypes.array
};
export default BrandPSRDetails;