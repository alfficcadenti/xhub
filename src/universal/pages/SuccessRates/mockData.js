export const successRatesData = [
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}]
];
