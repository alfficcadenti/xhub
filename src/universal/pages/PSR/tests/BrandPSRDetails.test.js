/* eslint-disable no-unused-vars */
import React from 'react';
import chai, {expect} from 'chai';
import {shallow, render} from 'enzyme';
import BrandPSRDetails from '../BrandPSRDetails';
import mockData from './data.test.json';
import chaiJestSnapshot from 'chai-jest-snapshot';
import renderer from 'react-test-renderer';
import h from '../psrHelpers';
chai.use(chaiJestSnapshot);

beforeEach(() => {
    chaiJestSnapshot.setFilename(`${__filename}.snap`);
});

describe('<BrandPSRDetails/>', () => {
    it('renders successfully', () => {
        const wrapper = shallow(<BrandPSRDetails />);
        expect(wrapper).to.have.length(1);
    });

    it('matches the snapshot', () => {
        chaiJestSnapshot.setTestName('matches the snapshot');
        const brandData = h.psrDetailsByBrand(mockData, 'egencia');
        const wrapper = render(<BrandPSRDetails data={brandData}/>);
        expect(wrapper).to.matchSnapshot();
    });
});