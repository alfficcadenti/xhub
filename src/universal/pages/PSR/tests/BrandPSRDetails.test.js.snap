// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  id="PSRDetails"
>
  <div>
    <table
      className="data-table"
    >
      <thead
        className="data-table-header"
      >
        <tr>
          <th
            className="pointer"
            onClick={[Function]}
          >
            Line Of Business
          </th>
          <th
            className="pointer"
            onClick={[Function]}
          >
            Last 24 hours
          </th>
          <th
            className="pointer"
            onClick={[Function]}
          >
            Last 7 days
          </th>
          <th
            className="pointer"
            onClick={[Function]}
          >
            Last 28 days
          </th>
        </tr>
      </thead>
      <tbody
        className="data-table-body"
      >
        <tr>
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "PSR",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "97.00 %",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "82.70 %",
              }
            }
          />
        </tr>
        <tr>
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "PSR Flights",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
          <td
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "85.00 %",
              }
            }
          />
        </tr>
      </tbody>
    </table>
    <div />
  </div>
</div>
`;
